featuretools/entityset/serialize.py

```diff
@@ -97,23 +104,65 @@ def write_entity_data(entity, path, format='csv', **kwargs):
     return {'location': location, 'type': format, 'params': kwargs}
 
 
-def write_data_description(entityset, path, **kwargs):
-    '''Serialize entityset to data description and write to disk.
+def write_data_description(entityset, path, profile_name=None, **kwargs):
+    '''Serialize entityset to data description and write to disk or S3 path.
 
     Args:
         entityset (EntitySet) : Instance of :class:`.EntitySet`.
-        path (str) : Location on disk to write `data_description.json` and entity data.
-        kwargs (keywords) : Additional keyword arguments to pass as keywords arguments to the underlying serialization method.
+        path (str) : Location on disk or S3 path to write `data_description.json` and entity data.
+        profile_name (str): The AWS profile specified to write to S3. Default: None
+        kwargs (keywords) : Additional keyword arguments to pass as keywords arguments to the underlying serialization method or to specify AWS profile.
     '''
-    path = os.path.abspath(path)
-    if os.path.exists(path):
-        shutil.rmtree(path)
-    for dirname in [path, os.path.join(path, 'data')]:
-        os.makedirs(dirname)
-    description = entityset_to_description(entityset)
-    for entity in entityset.entities:
-        loading_info = write_entity_data(entity, path, **kwargs)
-        description['entities'][entity.id]['loading_info'].update(loading_info)
-    file = os.path.join(path, 'data_description.json')
-    with open(file, 'w') as file:
-        json.dump(description, file)
+    if _is_s3(path):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            os.makedirs(os.path.join(tmpdir, 'data'))
+            description = entityset_to_description(entityset)
+            for entity in entityset.entities:
+                loading_info = write_entity_data(entity, tmpdir, **kwargs)
+                description['entities'][entity.id]['loading_info'].update(loading_info)
+            file = os.path.join(tmpdir, 'data_description.json')
+            with open(file, 'w') as file:
+                json.dump(description, file)
+
+            file_name = "es-{date:%Y-%m-%d_%H:%M:%S}".format(date=datetime.datetime.now())
+            file_path = os.path.join(tmpdir, file_name)
+            tar = tarfile.open(str(file_path) + ".tar", 'w')
+            tar.add(str(tmpdir) + '/data_description.json', arcname='data_description.json')
+            tar.add(str(tmpdir) + '/data', arcname='/data')
```

Comment:
why does this arcname begin with a forward slash but the other does not?

Reply:
The forward slash is used to create a directory called data to store the serialized data files whereas data_description.json is just a file.